### ISSUE TITLE
Wrong error report when macro in macro

### ISSUE THREAD

**Issue by NichtsVonChaos** [rust-analyzer/rust-analyzer#7039]
Write some code like:
```rust
use serde_json::json;

fn main() {
    let a = 1;
    let b = json!(format!("{}", a));
}
````
Obviously it can be compiled successfully.
But RA will report:
```
could not resolve macro `$crate::format`
```

**Comment by bjorn3** [comment]
Duplicate of #6788
